### PR TITLE
PROTOS: this change adds on_error conditions to PipelineStep + metadata

### PR DESCRIPTION
**Pull request by dselans**
This adds `PipelineStep.on_error` - that is to be read/used by SDK when a wasm returns a wasmresponse with an internal failure for exit code.

Also changed up conditions - no more `repeated Conditions` - it is now just a message type. This way there is no way to add "abort_current" and "abort_all" - they should be mutually exclusive. With this change, they are.

Last part is metadata. I will explain it in standup or you can read the updated spec: https://www.notion.so/streamdal/SPEC-Client-SDK-v2-80dafe668c524be7b0d944a7ae9ed92a?pvs=4#4d2c3f895d804c37a6a225cf8e2f4765

Short version: metadata is a step condition like notify or abort_all - it allows the user to say "when an error occurs, include this metadata in the response from the SDK". Meaning, the caller can then interpret the metadata as they wish.

Idea is to use this to facilitate drop_message.

LAST: I removed all "drop_message"-related things as they are replaced by metadata.

### CODE REVIEW

**Comment by dselans** [comment]
Going to redo this as a new PR. Mark had a lot of changes and I'm afraid resolving conflicts will screw something up.
